cpp/tests/c_api/ecg_test.c

```diff
@@ -0,0 +1,205 @@
+  data_type_id_t edge_id_tid   = INT32;
+  data_type_id_t edge_type_tid = INT32;
+
+  handle = cugraph_create_resource_handle(NULL);
+  TEST_ASSERT(test_ret_value, handle != NULL, "resource handle creation failed.");
+
+  ret_code = create_sg_test_graph(handle,
+                                  vertex_tid,
+                                  edge_tid,
+                                  h_src,
+                                  h_dst,
+                                  weight_tid,
+                                  h_wgt,
+                                  edge_type_tid,
+                                  NULL,
+                                  edge_id_tid,
+                                  NULL,
+                                  num_edges,
+                                  store_transposed,
+                                  FALSE,
+                                  FALSE,
+                                  FALSE,
+                                  &graph,
+                                  &ret_error);
+
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "create_test_graph failed.");
+  TEST_ALWAYS_ASSERT(ret_code == CUGRAPH_SUCCESS, cugraph_error_message(ret_error));
+
+  cugraph_rng_state_t* rng_state;
+  ret_code = cugraph_rng_state_create(handle, 0, &rng_state, &ret_error);
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "rng_state create failed.");
+
+  ret_code = cugraph_ecg(handle,
+                         rng_state,
+                         graph,
+                         min_weight,
+                         ensemble_size,
+                         max_level,
+                         threshold,
+                         resolution,
+                         FALSE,
+                         &result,
+                         &ret_error);
+
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, cugraph_error_message(ret_error));
+  TEST_ALWAYS_ASSERT(ret_code == CUGRAPH_SUCCESS, "cugraph_ecg failed.");
+
+  if (test_ret_value == 0) {
+    cugraph_type_erased_device_array_view_t* vertices;
+    cugraph_type_erased_device_array_view_t* clusters;
+
+    vertices          = cugraph_hierarchical_clustering_result_get_vertices(result);
+    clusters          = cugraph_hierarchical_clustering_result_get_clusters(result);
+    double modularity = cugraph_hierarchical_clustering_result_get_modularity(result);
+
+    vertex_t h_vertices[num_vertices];
+    edge_t h_clusters[num_vertices];
+
+    ret_code = cugraph_type_erased_device_array_view_copy_to_host(
+      handle, (byte_t*)h_vertices, vertices, &ret_error);
+    TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "copy_to_host failed.");
+
+    ret_code = cugraph_type_erased_device_array_view_copy_to_host(
+      handle, (byte_t*)h_clusters, clusters, &ret_error);
+    TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "copy_to_host failed.");
+
+    // Louvain and permute_range are both tested, here we only make
```

Comment:
I got the answer for my above comment here. But in this case, isn't it better to not pass arguments to `generic_ecg_test` that are not going to be used like `h_result`?

Reply:
Just a question and ECG always use multiple Louvain runs or it can use either Louvain or Leiden?

Reply:
At the moment it's only Louvain.